druid/src/core.rs

```diff
@@ -473,17 +473,29 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
                 r
             }
             LifeCycle::WidgetAdded => {
-                // if this is called, it means widgets were added. That might be
-                // be us; if we don't have old_data we set it now.
+                assert!(self.old_data.is_none());
```

Comment:
happy to see this, I'm  not 100% confident about some of the logic either so it's good to keep ourselves honest.